storage/innobase/xtrabackup/test/t/innodb_keyring_file_component.sh

```diff
@@ -0,0 +1,61 @@
+#
+# Component keyring file specific variables
+#
+require_server_version_higher_than 8.0.23
+
+KEYRING_TYPE="component"
+. inc/keyring_common.sh
+. inc/keyring_file.sh
+configure_server_with_component
+
+run_cmd $MYSQL $MYSQL_ARGS test <<EOF
+CREATE TABLE t1 (c1 VARCHAR(100)) ENCRYPTION='y';
+INSERT INTO t1 (c1) VALUES ('ONE'), ('TWO'), ('THREE');
+INSERT INTO t1 (c1) VALUES ('10'), ('20'), ('30');
+INSERT INTO t1 SELECT * FROM t1;
+INSERT INTO t1 SELECT * FROM t1;
+INSERT INTO t1 SELECT * FROM t1;
+INSERT INTO t1 SELECT * FROM t1;
+ALTER INSTANCE ROTATE INNODB MASTER KEY;
+CREATE TABLE t2 (c1 VARCHAR(100)) ENCRYPTION='y';
+INSERT INTO t2 SELECT * FROM t1;
+EOF
+
+xtrabackup --backup --target-dir=$topdir/backup1
+cp -r $topdir/backup1 $topdir/backup2
+record_db_state test
+shutdown_server
+
+xtrabackup --stats --datadir=${mysql_datadir} \
+  --component-keyring-file-config=${MYSQLD_DATADIR}/component_keyring_file.cnf \
+  --xtrabackup-plugin-dir=${plugin_dir}
+
+if grep -q "Encryption can't find master key, please check the keyring is loaded." $OUTFILE
+then
+  die "Cannot read component keyring file config to run stats on live server"
+fi
+
+# Test 1: Try to prepare with rubish on component config on target-dir
+cp $topdir/backup1/xtrabackup_component_keyring_file.cnf $topdir
+echo "bla" > $topdir/backup1/xtrabackup_component_keyring_file.cnf
+run_cmd_expect_failure $XB_BIN $XB_ARGS \
+--prepare --target-dir=$topdir/backup1 --xtrabackup-plugin-dir=${plugin_dir}
+
+# Test 2: Try to prepare with good component config on target-dir but
+# pass rubish as parameter
+run_cmd_expect_failure $XB_BIN $XB_ARGS \
+--prepare --target-dir=$topdir/backup2 \
+--component-keyring-file-config=$topdir/backup1/xtrabackup_component_keyring_file.cnf \
+--xtrabackup-plugin-dir=${plugin_dir}
```

Comment:
is there in difference in ${plugin_dir} vs $plugin_dir?

Reply:
@satya-bodapati no, they are the same. Just in case you have like:
var=foo
and want to print $var + the string bar you cannot do $varbar you will have to do ${var}bar